pkg/virt-launcher/virtwrap/converter/converter_test.go

```diff
@@ -1709,24 +1732,37 @@ var _ = Describe("Converter", func() {
 			Expect(domain.Spec.Devices.Video).To(HaveLen(devices))
 			Expect(domain.Spec.Devices.Graphics).To(HaveLen(devices))
 
-			if isARM64(arch) && (autoAttach == nil || *autoAttach) {
-				Expect(domain.Spec.Devices.Video[0].Model.Type).To(Equal(v1.VirtIO))
-				Expect(domain.Spec.Devices.Inputs[0].Type).To(Equal(v1.InputTypeTablet))
-				Expect(domain.Spec.Devices.Inputs[1].Type).To(Equal(v1.InputTypeKeyboard))
-			}
-			if isAMD64(arch) && (autoAttach == nil || *autoAttach) {
-				Expect(domain.Spec.Devices.Video[0].Model.Type).To(Equal("vga"))
+			if autoAttach == nil || *autoAttach {
+				switch arch {
+				case "amd64":
+				case "ppc64le":
+					Expect(domain.Spec.Devices.Video[0].Model.Type).To(Equal("vga"))
+					Expect(domain.Spec.Devices.Inputs).To(BeEmpty())
+				case "arm64":
+					Expect(domain.Spec.Devices.Video[0].Model.Type).To(Equal(v1.VirtIO))
+					Expect(domain.Spec.Devices.Inputs[0].Type).To(Equal(v1.InputTypeTablet))
+					Expect(domain.Spec.Devices.Inputs[1].Type).To(Equal(v1.InputTypeKeyboard))
+				case "s390x":
+					Expect(domain.Spec.Devices.Video[0].Model.Type).To(Equal(v1.VirtIO))
+					Expect(domain.Spec.Devices.Inputs).To(BeEmpty())
+				}
 			}
 		},
 			Entry("and add the graphics and video device if it is not set on amd64", nil, 1, "amd64"),
```

Comment:
I see these Cartesian product esque tables repeating a lot in converter tests due to testing every arch for the same cases.

I wonder if something like:
```go
			func() []TableEntry {
				var entries []TableEntry
				for _, arch := range []string{"amd64", "arm64", "ppc64le", "s390x"} {
					entries = append(entries,
						Entry(fmt.Sprintf("and add the graphics and video device if it is not set on %s", arch), nil, 1, arch),
						Entry(fmt.Sprintf("and add the graphics and video device if it is set to true on %s", arch), pointer.P(true), 1, arch),
						Entry(fmt.Sprintf("and not add the graphics and video device if it is set to false on %s", arch), pointer.P(false), 0, arch),
					)
				}
				return entries
			}(),
```

Or
```go
	for _, arch := range []string{"amd64", "arm64", "ppc64le", "s390x"} {
		Describe(fmt.Sprintf("on architecture %s", arch), func() {
			DescribeTable("should check autoAttachGraphicsDevices",
				func(autoAttach *bool, devices int) {
					...
				},
				Entry("and add the graphics and video device if it is not set", nil, 1),
				Entry("and add the graphics and video device if it is set to true", pointer.P(true), 1),
				Entry("and not add the graphics and video device if it is set to false", pointer.P(false), 0),
			)
		})
	}
```

Would be better, WDYT?

Reply:
Thank you! I think it's a great idea :) But I propose it in a different way:
```
func MultiArchEntry(text string, args ...interface{}) []TableEntry {
	return []TableEntry{
		Entry(fmt.Sprintf("%s on %s", text, "amd64"), append([]interface{}{"amd64"}, args...)...),
		Entry(fmt.Sprintf("%s on %s", text, "arm64"), append([]interface{}{"arm64"}, args...)...),
		Entry(fmt.Sprintf("%s on %s", text, "ppc64le"), append([]interface{}{"ppc64le"}, args...)...),
		Entry(fmt.Sprintf("%s on %s", text, "s390x"), append([]interface{}{"s390x"}, args...)...),
	}
}
```
https://github.com/kubevirt/kubevirt/pull/13300/commits/5c6bfc8109798eaf5c3b39f54072a90a5a39f858
WDYT?

Reply:
Awesome, do note that it's a bit limiting due to the strict format of both the description and subsequent args (need to ensure that the table's parameters are called in apt order) but looks great.